types/regexp.prototype.flags/regexp.prototype.flags-tests.ts

```diff
@@ -0,0 +1,20 @@
+import flags = require("regexp.prototype.flags");
+import "regexp.prototype.flags/auto";
+
+type D = 'd' | '';
+type G = 'g' | '';
+type I = 'i' | '';
+type M = 'm' | '';
+type S = 's' | '';
+type U = 'u' | '';
+type V = 'v' | '';
+type Y = 'y' | '';
```

Comment:
This all won't really work in a test. I would just name your type in implementation.d.ts.

Reply:
hmm, for CJS compat i need `export =`, so i can't export a separate type. i tried to get at it with `ReturnType<>` but it doesn't work.

Reply:
You can using:

```ts
namespace foo {
    type Flags = ...;
}

function foo(): foo.Flags;

export = foo;
```

Reply:
hm, ok - and on the consuming side, i can `import type { Flags } from` it?

Reply:
There's an example in the readme; I'd link it but I'm on my phone :P

Reply:
Yeah, downstream with esModuleInterop, but on DT tests you'd do a require and then fully qualify it.

Reply:
thanks! i'll see what i can do.

Reply:
this seems to work